lib/compat/wordpress-6.1/class-wp-theme-json-resolver-6-1.php

```diff
@@ -56,8 +56,50 @@ public static function get_core_data() {
 
 		$config       = static::read_json_file( __DIR__ . '/theme.json' );
 		$config       = static::translate( $config );
-		static::$core = new WP_Theme_JSON_Gutenberg( $config, 'default' );
+		$config       = apply_filters( 'global_styles_default', new WP_Theme_JSON_Data( $config, 'default' ) );
+		static::$core = new WP_Theme_JSON_Gutenberg( $config->get_data(), 'default' );
```

Comment:
With WP filters we need to return the same data type as in the input, so that should be `$data` instead of config, so maybe the following would be more accurate:
```php
$data = apply_filters( 'global_styles_default', new WP_Theme_JSON_Data( $config, 'default' ) );
 static::$core = new WP_Theme_JSON_Gutenberg( $data->get_config(), 'default' );
```